CodeEdit/Features/Settings/SettingsView.swift

```diff
@@ -7,50 +7,123 @@
+            } else if item.name.rawValue.contains(lowercasedSearchText) && !item.isSetting {
+                foundPage = true
+            }
+        }
+
+        return SettingsSearchResult(pageFound: foundPage, pages: returnedPages)
+    }
+
+    /// Creates a SettingsPage and it's respective child settings
+    private func createPageAndSettings(
+        _ settingsStruct: Any,
+        parent: SettingsPage,
+        prePages: [SettingsPage]
+    ) -> [SettingsPage] {
+        var pages: [SettingsPage] = prePages
+        pages.append(parent)
+
+        for setting in SettingsData().propertiesOf(settingsStruct) {
+            pages.append(
+                SettingsPage(
+                    parent.name,
+                    baseColor: parent.baseColor,
+                    icon: parent.icon,
+                    isSetting: true,
+                    displayName: setting.nameString
+                )
+            )
+        }
+
+        return pages
+    }
+
+    /// Creates all the neccessary pages
+    private func populatePages() -> [SettingsPage] {
+        var pages = [SettingsPage]()
+        let settingsData = SettingsData()
+
+        let generalSettings = SettingsPage(.general, baseColor: .gray, icon: .system("gear"))
+        pages = createPageAndSettings(settingsData.general, parent: generalSettings, prePages: pages)
+
+        let accountsSettings = SettingsPage(.accounts, baseColor: .blue, icon: .system("at"))
+        pages = createPageAndSettings(settingsData.accounts, parent: accountsSettings, prePages: pages)
+
+        let themesSettings = SettingsPage(.theme, baseColor: .pink, icon: .system("paintbrush.fill"))
+        pages = createPageAndSettings(settingsData.theme, parent: themesSettings, prePages: pages)
+
+        let textEditingSettings = SettingsPage(.textEditing, baseColor: .blue, icon: .system("pencil.line"))
+        pages = createPageAndSettings(settingsData.textEditing, parent: textEditingSettings, prePages: pages)
+
+        let terminalSettings = SettingsPage(.terminal, baseColor: .blue, icon: .system("terminal.fill"))
+        pages = createPageAndSettings(settingsData.terminal, parent: terminalSettings, prePages: pages)
+
+        let sourceControlSettings = SettingsPage(.sourceControl, baseColor: .blue, icon: .symbol("vault"))
+        pages = createPageAndSettings(settingsData.sourceControl, parent: sourceControlSettings, prePages: pages)
+
+        let locationsSettings = SettingsPage(.location, baseColor: .green, icon: .system("externaldrive.fill"))
+        pages = createPageAndSettings(settingsData.sourceControl, parent: locationsSettings, prePages: pages)
+
+        let featureFlags = SettingsPage(.featureFlags, baseColor: .cyan, icon: .system("flag.2.crossed.fill"))
+        pages = createPageAndSettings(settingsData.featureFlags, parent: featureFlags, prePages: pages)
+
+        return pages
+    }
+
     var body: some View {
+        let pages: [SettingsPage] = populatePages()
```

Comment:
This is re-generated every view update, which is wasting resources. Lets keep the static property of pages if possible.

Reply:
Fixed